core/dataset.cpp

```diff
@@ -703,6 +708,28 @@ bool DatasetConstProxy::operator!=(const DatasetConstProxy &other) const {
   return !dataset_equals(*this, other);
 }
 
+std::unordered_map<Dim, scipp::index> DatasetConstProxy::dimensions(
+    const std::vector<std::pair<Slice, scipp::index>> &currentSlices) const {
+
+  auto base_dims = m_dataset->dimensions();
+  // Note current slices are ordered, but NOT unique
+  for (const auto & [ slice, extents ] : currentSlices) {
+    (void)extents;
+    auto it = base_dims.find(slice.dim);
+    if (it != base_dims.end()) {
```

Comment:
Isn't the whole point of this PR to ensure that `it` will always be valid, i.e., you can drop this check?

Reply:
Indeed you should alway have a subset of the `Dataset` superset.